src/data/finset/powerset.lean

```diff
@@ -67,8 +67,63 @@ begin
     simp [finset.erase_eq_of_not_mem h, this] }
 end
 
+/-- For predicate `p` decidable on subsets, it is decidable whether `p` holds for any subset. -/
+def decidable_exists_of_decidable_subsets {s : finset α} {p : ∀ t ⊆ s, Prop}
+  (hu : ∀ t (h : t ⊆ s), decidable (p t h)) : decidable (∃ t (h : t ⊆ s), p t h) :=
```

Comment:
I wasn't able to channel the proof of `∀ t (h : t ⊆ s), decidable (p t h)` through the typeclass system. Eventually I decided that because of the dependence on `s` it couldn't be done and the proof had to be provided explicitly.